### ISSUE TITLE
Terrain shifted

### ISSUE THREAD

**Issue by DavixDevelop** [orangeadam3/terra121#114]
I tested this on both bteairoecean and the default projection. When teleporting to 46.221099, 15.266406, the water and the roads are placed correctly, but it seems like the terrain itself is shifted. For example, in the screenshot below, the river and the road is positioned on the left in the valley in real life, but here it's placed uphill.
![](https://user-images.githubusercontent.com/12420076/79511714-27aaf900-8040-11ea-9f35-37c2885afee8.png)
Another example at the same location.
![](https://user-images.githubusercontent.com/12420076/79511816-55903d80-8040-11ea-929f-de9ac7fa8360.png)
I checked the dem files itself and overlayed it with the default map in osm, and it's aligned correctly, so the dem files itself are correct.

**Comment by taeko-chan** [comment]
ok correct me if i'm wrong, but you're saying the river in the top picture is supposed to flow to the left of that hill and so is the road?

**Comment by DavixDevelop** [comment]
Yes. This is how it looks like in real life. The hill starts right at the edge of the road, like in the bottom picture on the left.
![](https://user-images.githubusercontent.com/12420076/79797658-ef652c80-8357-11ea-815d-a27990c02cdb.jpg)
Another example.
![](https://user-images.githubusercontent.com/12420076/79797829-42d77a80-8358-11ea-90b9-c7f6677069e8.jpg)

**Comment by taeko-chan** [comment]
the way the roads and rivers are generated is by estimating the height of the terrain at a given x and z (converted from the latitude and longitude from the OSM data) and placing it there. obviously that creates problems. for example, here. i'm not sure exactly what the problem is, but the mod owner (@orangeadam3) is working on a bug he calls grounding issues, and this may be covered by that, so i'll let you know when a fix for that comes out. if the problem persists afterwards, i'll see what i can do about it.
